activerecord/lib/active_record/type/decimal_without_scale.rb

```diff
@@ -10,6 +10,22 @@ def type
       def type_cast_for_schema(value)
         value.to_s.inspect
       end
+
+      def serialize(value)
+        raise ActiveModel::RangeError, "cannot be infinite" if value&.to_d&.infinite?
+        super
+      end
+
+      private
+
+        def cast_value(value)
+          case value
+          when true then 1
+          when false then 0
+          else
+            value.to_d rescue nil
```

Comment:
This opens up a can of worms 😅

The `DecimalWithoutScale` type intentionally casts to Integer, not BigDecimal. The idea is to let you store large Integer values that won't fit in the database's native INTEGER type in a DECIMAL column instead, and have them mapped back to Integer values again when the record is loaded from the database.

https://github.com/rails/rails/blob/ab5d5c95ec51c63d58d7e75694af748f165e2228/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L822

I admit that I find this behaviour unintuitive, but it's covered by tests, and changing it would likely break users' applications. My suggestion is to leave the `DecimalWithoutScale` class unchanged here; it will mean that infinite values are treated differently by decimal attributes without a scale, but they already treat _every other value_ differently, so the inconsistency is unlikely to cause problems in practice.